src/control/settings/Settings.h

```diff
@@ -136,6 +145,7 @@ class Settings {
 
     void saveData(xmlNodePtr root, const string& name, SElement& elem);
 
+    static const char* getButtonConfigNames(int button);
```

Comment:
The function is only used in the sourcefile, so no explicit forward declaration is necessary.

Reply:
Sorry, I am not following you (as I said my C++ is rusty). How can I define it as a class member if not in the class declaration? Are you suggesting it should be a standalone function within the cpp file rather than a static class member? If that's the case, I am not sure I agree because semantically it's part of the class (any derived class would need to be aware of such mapping and possibly be able to use it, without having to "include" the cpp file).

Reply:
@lciti 
 > Are you suggesting it should be a standalone function within the cpp file rather than a static class member?

Yes it should be a standalone function in the source file.

 > If that's the case, I am not sure I agree because semantically it's part of the class (any derived class would need to be aware of such mapping and possibly be able to use it, without having to "include" the cpp file).

Yet no class derives from this and it is less likely this will happen. Also inheritance should be avoided in 99% of any cases. Also if that's the case the complete function can be moved into a header.
I also think the Settings class is not an EnumMapper, so basically this function belongs right to the Enum definition as free function, if used at many places.

Reply:
constexpr functions can't be split above source and header files, when they are used in many compilation units, because they must be available for each using compilation unit on compile time.
You would write such function like:   
`constexpr MyType my_func(Input in) {...}`   
When it is static:   
`constexpr static MyType my_func(Input in) {...}`

Reply:
Sorry, I missed your replies as I had not updated the page

Reply:
> I also think the Settings class is not an EnumMapper, so basically this function belongs right to the Enum definition as free function, if used at many places.

That's a good point. If I understand correctly, you are saying the function should be in the header right after the definition of the corresponding enum.

Reply:
Exactly 😬.